server/sonar-server/src/test/java/org/sonar/server/settings/ChildSettingsTest.java

```diff
@@ -0,0 +1,124 @@
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.server.settings;
+
+import java.util.Collections;
+import java.util.Date;
+import java.util.Optional;
+import java.util.Random;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.sonar.api.config.PropertyDefinition;
+import org.sonar.api.config.PropertyDefinitions;
+import org.sonar.api.config.internal.MapSettings;
+
+import static org.apache.commons.lang.RandomStringUtils.randomAlphanumeric;
+import static org.assertj.core.api.Java6Assertions.assertThat;
+
+public class ChildSettingsTest {
+  private static final Random RANDOM = new Random();
+
+  @Rule
+  public ExpectedException expectedException = ExpectedException.none();
+
+  private MapSettings parent = new MapSettings();
+  private ChildSettings underTest = new ChildSettings(parent);
+
+  @Test
+  public void childSettings_should_retrieve_parent_settings() {
+    String multipleValuesKey = randomAlphanumeric(19);
+    PropertyDefinition multipleValues = PropertyDefinition.builder(multipleValuesKey).multiValues(true).build();
+    MapSettings parent = new MapSettings(new PropertyDefinitions(Collections.singletonList(multipleValues)));
+    ChildSettings underTest = new ChildSettings(parent);
+
+    parent.setProperty(randomAlphanumeric(10), randomAlphanumeric(20));
+    parent.setProperty(randomAlphanumeric(11), RANDOM.nextLong());
+    parent.setProperty(randomAlphanumeric(12), RANDOM.nextDouble());
+    parent.setProperty(randomAlphanumeric(13), RANDOM.nextFloat());
+    parent.setProperty(randomAlphanumeric(14), RANDOM.nextBoolean());
+    parent.setProperty(randomAlphanumeric(15), RANDOM.nextInt());
+    parent.setProperty(randomAlphanumeric(16), new Date(RANDOM.nextInt()));
+    parent.setProperty(randomAlphanumeric(17), new Date(RANDOM.nextInt()), true);
+    parent.setProperty(randomAlphanumeric(18), new Date(RANDOM.nextInt()), false);
+    parent.setProperty(multipleValuesKey, new String[] { randomAlphanumeric(10), randomAlphanumeric(20) });
+
+    assertThat(underTest.getProperties()).isEqualTo(parent.getProperties());
+  }
+
+  @Test
+  public void set_will_throw_NPE_if_key_is_null() {
+    expectedException.expect(NullPointerException.class);
+    expectedException.expectMessage("key can't be null");
+
+    underTest.set(null, "");
+  }
+
+
+  @Test
+  public void set_will_throw_NPE_if_value_is_null() {
+    expectedException.expect(NullPointerException.class);
+    expectedException.expectMessage("value can't be null");
+
+    underTest.set(randomAlphanumeric(10), null);
+  }
+
+  @Test
+  public void childSettings_override_parent() {
+    String key = randomAlphanumeric(10);
+    parent.setProperty(key, randomAlphanumeric(20));
+    underTest.setProperty(key, randomAlphanumeric(10));
+
+    assertThat(underTest.get(key)).isNotEqualTo(parent.getString(key));
```

Comment:
You could add check, that it actually *is*  equal to the string that you set to `underTest`.